gcore/gdaldataset.cpp

```diff
@@ -2765,6 +2765,16 @@ CPLErr GDALDataset::RasterIO(GDALRWFlag eRWFlag, int nXOff, int nYOff,
         }
     }
 
+#ifndef HAVE__FLOAT16
+    if (eBufType == GDT_Float16 || eBufType == GDT_CFloat16)
```

Comment:
We shouldn't error out if we declare datasets with Float16 data type when HAVE__FLOAT16 is not defined (which is currently the case in the current state of this PR for the Zarr driver)

Reply:
Correct, this was an oversight.